integration/test/migration/main_test.go

```diff
@@ -0,0 +1,168 @@
+// +build k8srequired
+
+package migration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/e2e-harness/pkg/framework/deployment"
+	"github.com/giantswarm/e2e-harness/pkg/framework/resource"
+	e2esetup "github.com/giantswarm/e2esetup/chart"
+	"github.com/giantswarm/e2esetup/chart/env"
+	"github.com/giantswarm/e2etests/managedservices"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kubernetes-metrics-server/integration/templates"
+)
+
+const (
+	chartName         = "kubernetes-metrics-server"
+	metricsServerName = "metrics-server"
+	testName          = "migration"
+)
+
+var (
+	a          *apprclient.Client
+	d          *deployment.Deployment
+	ms         *managedservices.ManagedServices
+	h          *framework.Host
+	helmClient *helmclient.Client
+	l          micrologger.Logger
+	r          *resource.Resource
+)
+
+func init() {
+	var err error
+
+	{
+		c := micrologger.Config{}
+		l, err = micrologger.New(c)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		c := apprclient.Config{
+			Fs:     afero.NewOsFs(),
+			Logger: l,
+
+			Address:      "https://quay.io",
+			Organization: "giantswarm",
+		}
+		a, err = apprclient.New(c)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		c := framework.HostConfig{
+			Logger:     l,
+			ClusterID:  "na",
+			VaultToken: "na",
+		}
+		h, err = framework.NewHost(c)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		c := deployment.Config{
+			K8sClient: h.K8sClient(),
+			Logger:    l,
+		}
+		d, err = deployment.New(c)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		c := helmclient.Config{
+			Logger:          l,
+			K8sClient:       h.K8sClient(),
+			RestConfig:      h.RestConfig(),
+			TillerNamespace: "giantswarm",
+		}
+		helmClient, err = helmclient.New(c)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
```

Comment:
This block should be removed.

The managed services test is only needed for the basic test. See https://github.com/giantswarm/kubernetes-node-exporter/blob/b7aec49844d9b6797fa647ab8758591729cf97f0/integration/test/migration/main_test.go for the wiring for a migration test.